src/Synapse/Synapse/Commands/DataPlaneCommands/AccessControl/GetAzureSynapseRoleAssignment.cs

```diff
@@ -0,0 +1,154 @@
+        [Parameter(ValueFromPipelineByPropertyName = false, ParameterSetName = GetByWorkspaceObjectAndRoleDefinitionIdAndObjectIdParameterSet,
+            Mandatory = true, HelpMessage = HelpMessages.RoleDefinitionId)]
+        [ValidateNotNullOrEmpty]
+        public string RoleDefinitionId { get; set; }
+
+        [Parameter(ValueFromPipelineByPropertyName = false, ParameterSetName = GetByWorkspaceNameAndNameParameterSet,
+            Mandatory = false, HelpMessage = HelpMessages.SignInName)]
+        [Parameter(ValueFromPipelineByPropertyName = false, ParameterSetName = GetByWorkspaceObjectAndNameParameterSet,
+            Mandatory = true, HelpMessage = HelpMessages.SignInName)]
+        [Alias("Email", "UserPrincipalName")]
+        [ValidateNotNullOrEmpty]
+        public string SignInName { get; set; }
+
+        [Parameter(ValueFromPipelineByPropertyName = false, ParameterSetName = GetByWorkspaceNameAndServicePrincipalNameParameterSet,
+            Mandatory = false, HelpMessage = HelpMessages.ServicePrincipalName)]
+        [Parameter(ValueFromPipelineByPropertyName = false, ParameterSetName = GetByWorkspaceObjectAndServicePrincipalNameParameterSet,
+            Mandatory = true, HelpMessage = HelpMessages.ServicePrincipalName)]
+        [ValidateNotNullOrEmpty]
+        public string ServicePrincipalName { get; set; }
+
+        [Parameter(ValueFromPipelineByPropertyName = false, ParameterSetName = GetByWorkspaceNameAndIdParameterSet,
+            Mandatory = false, HelpMessage = HelpMessages.PrincipalId)]
+        [Parameter(ValueFromPipelineByPropertyName = false, ParameterSetName = GetByWorkspaceObjectAndIdParameterSet,
+            Mandatory = false, HelpMessage = HelpMessages.PrincipalId)]
+        [Parameter(ValueFromPipelineByPropertyName = false, ParameterSetName = GetByWorkspaceNameAndRoleDefinitionIdAndObjectIdParameterSet,
+            Mandatory = false, HelpMessage = HelpMessages.PrincipalId)]
+        [Parameter(ValueFromPipelineByPropertyName = false, ParameterSetName = GetByWorkspaceObjectAndRoleDefinitionIdAndObjectIdParameterSet,
+            Mandatory = false, HelpMessage = HelpMessages.PrincipalId)]
+        [Alias("Id", "PrincipalId")]
+        [ValidateNotNullOrEmpty]
+        public string ObjectId { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            if (this.IsParameterBound(c => c.WorkspaceObject))
+            {
+                this.WorkspaceName = this.WorkspaceObject.Name;
+            }
+
+            if (this.IsParameterBound(c => c.RoleDefinitionName))
+            {
+                this.RoleDefinitionId = SynapseAnalyticsClient.GetRoleDefinitionIdFromRoleDefinitionName(this.RoleDefinitionName);
+            }
+
+            if (this.IsParameterBound(c => c.SignInName))
+            {
+                this.ObjectId = SynapseAnalyticsClient.GetObjectIdFromSignInName(this.SignInName);
+            }
+
+            if (this.IsParameterBound(c => c.ServicePrincipalName))
```

Comment:
>if (this.IsParameterBound(c => c.ServicePrincipalName)) [](start = 12, length = 55)

The same issue with above.

Reply:
I think that if ` if (this.IsParameterBound(c => c.xxx))` is related with the parameter `xx` support pipeline. If the parameter doesn't pipeline we don't need this line either. And we don't need to specify `ValueFromPipelineByPropertyName = false` explictly, the default value is `false`.
If I am wrong, Could you please fix me? @idear1203

---
In reply to: [448952444](https://github.com/Azure/azure-powershell/pull/12309#discussion_r448952444) [](ancestors = 448952444)

Reply:
As we discussed offline. I think they don't have strong relationship.